server/service/src/sync/translations/invoice_line.rs

```diff
@@ -275,6 +276,14 @@ impl SyncTranslation for InvoiceLineTranslation {
             inventory_adjustment_reason_id,
         } = InvoiceLineRowRepository::new(connection).find_one_by_id(&changelog.record_id)?;
 
+        // The item_id from RequisitionLineRow is actually for an item_link_id, so we get the true item_id here
+        let item_id = ItemLinkRowRepository::new(connection)
+            .find_one_by_id(&item_id)?
+            .ok_or(anyhow::anyhow!(
+                "Item ({item_id}) not found for invoice line ({id})"
+            ))?
+            .item_id;
```

Comment:
Shouldn't you use the returned item from the `InvoiceLineRepository::query()`?

Probably applies for previous PR... If possible using ItemLinkRowRepository should be avoided to be used outside of the repository layer...

Reply:
Pls feel free to request changes if something doesn't seem right 😀

> Probably applies for previous PR... 

Yea I did copy it from my `requisition_line` implementation. For this PR you're right I can use `InvoiceLineRepository::query()`, thanks!

> If possible using ItemLinkRowRepository should be avoided to be used outside of the repository layer...

Hey those are my words! 😆 Very fair. The reason I did it for `requisition_line` was because `RequisitionLineRowRepository` only returns `( RequisitionRow, RequisitionLineRow )` 🙁. I coooooould make it return the item (via link) but was avoiding the changes because I was worried how far they'd propagate heh. What do you reckon?

Reply:
Yeah think its ok to leave it for the requisition case